TS SDK: @solana/kora main entry pulls @solana/kit-plugin-* peers, breaking bundlers

## Problem
`import { KoraClient } from "@solana/kora"` (0.3.0-beta.x) fails to bundle with esbuild (and likely other bundlers):

```
✘ [ERROR] Could not resolve "@solana/kit-plugin-payer"
    .../@solana/kora/dist/src/kit/index.js:3:22
```

Consumers that only use `KoraClient` shouldn't need the `@solana/kit-plugin-*` peer deps at all.

## Root cause
`sdks/ts/src/index.ts` (the main barrel) re-exports the kit integration:
```ts
export { createKitKoraClient, kora, type KoraKitClient } from './kit/index.js';
```
`./kit/index.ts` imports `@solana/kit-plugin-instruction-plan`, `@solana/kit-plugin-rpc`, `@solana/kit-plugin-signer`. Because the barrel re-exports it, **every** `@solana/kora` import drags those peers into the graph, and bundlers fail to resolve them. `package.json` also has no `exports` map (`exports: null`), so the subpath can't be imported cleanly either.

## Impact
Broke the SDP API (`solana-developer-platform`) Docker/esbuild build when adopting 0.3.0-beta to use the new typed `user_id`. Worked around by staying on 0.2.1 + a request passthrough; this issue tracks the upstream fix so consumers can adopt 0.3.x cleanly.

## Fix
- Stop re-exporting the kit integration from the main barrel.
- Expose it on a `@solana/kora/kit` subpath via a proper `exports` map (`.` = core client, `./kit` = kit integration).
- `KoraClient` / `koraPlugin` / types stay on the main entry (bundle-clean).

Breaking for `0.3.0-beta` consumers of `createKitKoraClient`/`kora`/`KoraKitClient` (import from `@solana/kora/kit`).

--- a/sdks/ts/package.json
+++ b/sdks/ts/package.json
@@ -1,10 +1,20 @@
 {
   "name": "@solana/kora",
-  "version": "0.3.0-beta.1",
+  "version": "0.3.0-beta.2",
   "description": "TypeScript SDK for Kora RPC",
   "main": "dist/src/index.js",
   "type": "module",
   "types": "dist/src/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./dist/src/index.d.ts",
+      "default": "./dist/src/index.js"
+    },
+    "./kit": {
+      "types": "./dist/src/kit/index.d.ts",
+      "default": "./dist/src/kit/index.js"
+    }
+  },
   "scripts": {
     "build": "tsc",
     "test": "jest",

--- a/sdks/ts/src/index.ts
+++ b/sdks/ts/src/index.ts
@@ -1,5 +1,4 @@
 export * from './error.js';
 export * from './types/index.js';
 export { KoraClient } from './client.js';
-export { createKitKoraClient, kora, type KoraKitClient } from './kit/index.js';
 export { koraPlugin, type KoraApi } from './plugin.js';

--- a/sdks/ts/test/integration.test.ts
+++ b/sdks/ts/test/integration.test.ts
@@ -28,7 +28,8 @@ import {
     tokenProgram,
 } from '@solana-program/token';
 
-import { KoraClient, kora } from '../src/index.js';
+import { kora } from '../src/kit/index.js';
+import { KoraClient } from '../src/index.js';
 import { runAuthenticationTests } from './auth-setup.js';
 import setupTestSuite from './setup.js';
 

